app/models/concerns/taxonomix.rb

```diff
@@ -12,6 +12,7 @@ module Taxonomix
              :through => TAXONOMY_JOIN_TABLE, :source => :taxonomy,
              :validate => false
     after_initialize :set_current_taxonomy
+    before_save :add_taxonomies_having_select_all_enabled
```

Comment:
@orrabin,

Previously I have added it for new_objects only. 
Later in discussion with @ares found that from web-ui user can't modify taxonomies as field will be disabled but using API/hammer, there might be possibility user can change taxonomy.